ts/Series/TiledWebMap/TiledWebMapSeries.ts

```diff
@@ -0,0 +1,934 @@
+/* *
+ *
+ *  (c) 2010-2023 Hubert Kozik, Kamil Musiałowski
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+/* *
+ *  Imports
+ *
+ * */
+
+import TiledWebMapSeriesOptions from './TiledWebMapSeriesOptions.js';
+import SeriesRegistry from '../../Core/Series/SeriesRegistry.js';
+import type PositionObject from '../../Core/Renderer/PositionObject';
+import TilesProvidersRegistry from '../../Maps/TilesProviders/TilesProvidersRegistry.js';
+import SVGElement from '../../Core/Renderer/SVG/SVGElement.js';
+import Chart from '../../Core/Chart/Chart.js';
+import type {
+    AnimationStepCallbackFunction
+} from '../../Core/Animation/AnimationOptions';
+
+const {
+    seriesTypes: {
+        map: MapSeries
+    }
+} = SeriesRegistry;
+
+import U from '../../Core/Utilities.js';
+
+const {
+    addEvent,
+    defined,
+    error,
+    merge,
+    pick
+} = U;
+
+/* *
+ *
+ *  Declarations
+ *
+ * */
+
+interface TileItem extends SVGElement {
+    posX?: number,
+    posY?: number,
+    originalURL?: string,
+    isActive?: boolean
+}
+
+interface TilesItem {
+    tiles: Record<string, TileItem>,
+    isActive: boolean,
+    howManyTiles: number,
+    actualTilesCount: number,
+    loaded: boolean
+}
+
+/**
+ * The series type
+ *
+ * @private
+ * @class
+ * @name Highcharts.seriesTypes.tiledwebmap
+ *
+ * @augments Highcharts.Series
+ */
+class TiledWebMapSeries extends MapSeries {
+
+    /* *
+     *
+     * Static properties
+     *
+     * */
+
+    /**
+     * A tiledwebmap series allows user to display dynamically joined individual
+     * images (tiles) and join them together to create a map.
+     *
+     * @sample maps/series-tiledwebmap/simple-demo-norway
+     *         Simple demo of data for Norway on TiledWebMap
+     * @sample maps/series-tiledwebmap/only-twm
+     *         OpenStreetMap demo
+     *
+     * @extends      plotOptions.map
+     * @excluding    affectsMapView, allAreas, allowPointSelect, animation,
+     * animationLimit, boostBlending, boostThreshold, borderColor, borderWidth,
+     * clip, color, colorAxis, colorByPoint, colorIndex, colorKey, colors,
+     * cursor, dashStyle, dataLabels, dataParser, dataURL, dragDrop,
+     * enableMouseTracking, findNearestPointBy, joinBy, keys, marker,
+     * negativeColor, nullColor, nullInteraction, onPoint, point,
+     * pointDescriptionFormatter, selected, shadow, showCheckbox,
+     * stickyTracking, tooltip, type
+     * @product      highmaps
+     * @optionparent plotOptions.tiledwebmap
+     */
+
+    public static defaultOptions: TiledWebMapSeriesOptions = merge(MapSeries.defaultOptions, {
+        states: {
+            inactive: {
+                enabled: false
+            }
+        },
+        zIndex: -1
```

Comment:
Is there a special reason for a fixed default z index while all other series types have `undefined`? I think this may cause problems for implementers trying to stack map layers the usual way, by either defining the order in the `series` array, or by the [index option](https://api.highcharts.com/highcharts/series.line.index).

Reply:
The purpose of this was to place the TWM series always below any other series. For example, if a user would declare the TWM series as second, third, the last, the TWM won't be above the previous series. Sure, we can delete this default option and let the TWM series always be first in configuration.

Demo: https://jsfiddle.net/BlackLabel/7awn2y9g/

Reply:
Ok, that's what I figured. I still think we should delete it so that the tiledwebmap series behaves like any other series. In addition, it may be beneficial to add a paragraph in the tiledwebmap docs article, that the `series` array can be thought of like _layers_ that are stacked on top of each other, and that the ordering of the layers are determined by source order, and can be overridden through the `index` option and the `zIndex` option.

On the same note - maybe we should set [legend.reversed](https://api.highcharts.com/highcharts/legend.reversed) in the demos that currently apply map layers, like the `twm-providers` demo? Then the layering in the legend is more in line with the actual z-index.